OSPSuite.Presentation.Importer/Presenters/UnitsEditorPresenter.cs

```diff
@@ -0,0 +1,106 @@
+﻿using OSPSuite.Presentation.Importer.Core;
+using OSPSuite.Presentation.Importer.Views;
+using OSPSuite.Presentation.Presenters;
+using OSPSuite.Core.Domain.UnitSystem;
+using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Utility.Extensions;
+
+namespace OSPSuite.Presentation.Importer.Presenters
+{
+   public class UnitsEditorPresenter : AbstractDisposablePresenter<IUnitsEditorView, IUnitsEditorPresenter>, IUnitsEditorPresenter
+   {
+      private Column _importDataColumn;
+      private IEnumerable<IDimension> _dimensions;
+      public string SelectedUnit { get; private set; }
+      private bool _canClose = false;
+
+      public UnitsEditorPresenter(IUnitsEditorView view) : base(view)
+      {
+      }
+
+      public bool Canceled => _view.Canceled;
+
+      public override bool CanClose => base.CanClose && _canClose;
+
+      public void ShowFor(Column importDataColumn, IEnumerable<IDimension> dimensions)
+      {
+         SelectedUnit = importDataColumn.SelectedUnit;
+         _importDataColumn = importDataColumn;
+         _dimensions = dimensions;
+
+         fillDimensions();
+         fillUnits();
+
+         View.SetParams(useDimensionSelector());
+         _view.Display();
+      }
+
+      public void SelectDimension(string dimension)
+      {
+         this.DoWithinExceptionHandler(() =>
+         {
+            SelectedUnit = _dimensions.First(d => d.Name == dimension).DefaultUnit.Name;
+            fillUnits();
+         });
+      }
+
+      public void SelectUnit(string unit)
+      {
+         this.DoWithinExceptionHandler(() =>
+         {
+            SelectedUnit = unit;
+         });
+      }
+
+      private bool useDimensionSelector()
+      {
+         return _dimensions.Count() > 1;
+      }
+
+      private void fillDimensions()
+      {
+         if (useDimensionSelector())
+            View.FillDimensionComboBox(_dimensions, selectedDimension.Name);
+         else
+            View.FillDimensionComboBox(new List<IDimension>(), "");
+      }
+
+      private void fillUnits()
+      {
+         if (useDimensionSelector())
+         {
+            View.FillUnitComboBox(selectedDimension.Units, SelectedUnit);
+         }
+
+         if (_dimensions == null || !_dimensions.Any())
+            return;
+
+         View.FillUnitComboBox(_dimensions.SelectMany(d => d.Units), SelectedUnit);
+      }
+
+      private IDimension findDimension()
+      {
+         foreach (var dimension in _dimensions)
```

Comment:
This looks suspicious here. We have loads of methods doing this type of stuff already. see dimensionFactory and all other dimension extensions

Reply:
Done